test/IntelFPGAMemoryAttributes.ll

```diff
@@ -146,25 +150,32 @@ entry:
   %var_ten10 = bitcast i32* %var_ten to i8*
   ; CHECK-LLVM: call void @llvm.var.annotation(i8* [[VAR10:%[a-zA-Z0-9_]+]], i8* getelementptr inbounds ([37 x i8], [37 x i8]* [[STR10]], i32 0, i32 0), i8* undef, i32 undef)
   call void @llvm.var.annotation(i8* %var_ten10, i8* getelementptr inbounds ([37 x i8], [37 x i8]* @.str.10, i32 0, i32 0), i8* getelementptr inbounds ([13 x i8], [13 x i8]* @.str.1, i32 0, i32 0), i32 11)
-  %9 = bitcast i32* %var_ten to i8*
-  call void @llvm.lifetime.end.p0i8(i64 4, i8* %9) #4
-  %10 = bitcast i32* %var_nine to i8*
+  %9 = bitcast i32* %var_eleven to i8*
+  call void @llvm.lifetime.start.p0i8(i64 4, i8* %9) #4
+  %var_eleven11 = bitcast i32* %var_eleven to i8*
+  ; CHECK-LLVM: call void @llvm.var.annotation(i8* [[VAR11:%[a-zA-Z0-9_]+]], i8* getelementptr inbounds ([46 x i8], [46 x i8]* [[STR11]], i32 0, i32 0), i8* undef, i32 undef)
+  call void @llvm.var.annotation(i8* %var_eleven11, i8* getelementptr inbounds ([46 x i8], [46 x i8]* @.str.11, i32 0, i32 0), i8* getelementptr inbounds ([13 x i8], [13 x i8]* @.str.1, i32 0, i32 0), i32 12)
+  %10 = bitcast i32* %var_eleven to i8*
```

Comment:
(it may sound a bit daunting) Please, consider rewriting the test to use a separate function for each attribute? This would make any subsequent changes far easier. A `TODO` and addressing this separately would be fine.

Reply:
Actually, it is a good idea to rewrite test this way, should be easier to support. I will left a `TODO` note